### ISSUE TITLE
[osf] .sps file rendering in Chinese

### ISSUE THREAD

**Issue by raklein** [CenterForOpenScience/osf.io#3014]
Steps to reproduce:
1. Navigate to this page to try to render the .sps file https://osf.io/zvp3s/

Expected:
1. Displays contents of .sps in plain text

Actual:
1. Appears to be displaying it in Chinese or a related language. Note that this does not happen for all .sps files (this specific .sps file is to create a graph, other files in the same component display correctly)
![screen shot 2015-06-08 at 4 57 40 pm](https://cloud.githubusercontent.com/assets/8781936/8045158/84b6c3ee-0dff-11e5-9ab2-9327da1da9c0.png)

**Comment by chennan47** [comment]
This is not Chinese actually. It's related to code transition.

**Comment by lbanner** [comment]
@chennan47  -- Do we know why this file displays in this way?

**Comment by chennan47** [comment]
The encoding and decoding is not working correctly.

**Comment by lbanner** [comment]
Our issue then, yes? Do you think an intern could tackle?

**Comment by chennan47** [comment]
It's a MFR problem. Better have someone familiar with MFR to deal with it.

**Comment by chrisseto** [comment]
Are sps files binary files?

**Comment by raklein** [comment]
No it's a script file for the stats program SPSS. Can open these files in a
text editor but see below for an example:

"GET

FILE='C:\Users\Vianello\Documents\Dropbox\Manylabs\Results\effectsizes.graphdata1.sav'.
DATASET NAME esdata.

DATASET ACTIVATE esdata.

***add overall variables***

 * compute Overalld=0.
 * if Study='allowedforbiddend' Overalld=1.96.
 * if Study='Anchoring1d' Overalld=1.17.
 * if Study='Anchoring2d' Overalld=1.79.
 * if Study='Anchoring3d' Overalld=2.23.
 * if Study='Anchoring4d' Overalld=2.42.
 * if Study='contactd' Overalld=.13.
 * if Study='flagd' Overalld=.03.
 * if Study='gainlossd' Overalld=.60.
 * if Study='gambfald' Overalld=.61.
 * if Study='IATd' Overalld=.53.
 * if Study='moneyd' Overalld=-.02.
 * if Study='quoted' Overalld=.32.
 * if Study='reciprocityd' Overalld=.30.
 * if Study='scalesd' Overalld=.49.
 * if Study='sunkd' Overalld=.27.
 * EXECUTE."

**Comment by samanehsan** [comment]
@mfraezz, mind looking into this MFR issue? Thanks!

**Comment by mfraezz** [comment]
I'm not as much of an expert on MFR as I was before it became MFRAAS, but this is the only .sps in that project that doesn't render properly. Bash tells me the files are encoded differently:
```bash
$ file Downloads/Graph.sps
Downloads/Graph.sps: Non-ISO extended-ASCII text, with CRLF line terminators
$ file Downloads/Order.effects.moderations.sps
Downloads/Order.effects.moderations.sps: ASCII text, with CRLF line terminators
$ file Downloads/ICCs.sps
Downloads/ICCs.sps: ASCII text, with CRLF line terminators
```

Non-ISO means it contains characters not in the normal ASCII range (31-128), which is likely the reason it's displaying incorrectly. Searching for characters outside of this range gives us one character 
-- 0x92, or `’`, on line 100 of the file (hexdump also can't render it, so it shows  `.`) :
```bash
$ hexdump -C Downloads/Graph.sps | ag '[89a-f][0-9a-f]'
[...]
00000c10  6f 2e 20 67 61 6d 62 6c  65 72 92 73 20 66 61 6c  |o. gambler.s fal|
[...]
```

In short, the file's author didn't [escape](http://www.spss-tutorials.com/escape-sequence/) a quote correctly, but we should probably still handle this sort of thing. It's rendered by the `code_pygments` extension, which deals with more filetypes than any other renderer -- the difficulty isn't in fixing this, but making sure the fix doesn't break anything else rendered by this extension. I'd say it's appropriate for an intern task, though it's very low-priority.

**Comment by chrisseto** [comment]
This is fixed. @GaryKriebel
